src/print_graph.cc

```diff
@@ -186,25 +192,27 @@ namespace detail {
 					} else {
 						task_label += "(deleted)";
 					}
-					task_subgraph_dot.emplace(
-					    tid, fmt::format("subgraph cluster_{}{{label=<<font color=\"#606060\">{}</font>>;color=darkgray;", tid, task_label));
+					task_subgraph_dot.emplace(tid,
+					    fmt::format("subgraph cluster_{}{{label=<<font color=\"#606060\">{}</font>>;color=darkgray;", local_to_global_id(tid), task_label));
 				}
 				task_subgraph_dot[tid] += print_vertex(*cmd);
 			} else {
 				main_dot += print_vertex(*cmd);
 			}
 
 			for(const auto& d : cmd->get_dependencies()) {
-				fmt::format_to(std::back_inserter(main_dot), "{}->{}[{}];", d.node->get_cid(), cmd->get_cid(), dependency_style(d));
+				fmt::format_to(std::back_inserter(main_dot), "{}->{}[{}];", local_to_global_id(d.node->get_cid()), local_to_global_id(cmd->get_cid()),
+				    dependency_style(d));
 			}
 
 			// Add a dashed line to the corresponding push
-			if(const auto apcmd = dynamic_cast<const await_push_command*>(cmd)) {
-				fmt::format_to(std::back_inserter(main_dot), "{}->{}[style=dashed color=gray40];", apcmd->get_source()->get_cid(), cmd->get_cid());
-			}
+			// if(const auto apcmd = dynamic_cast<const await_push_command*>(cmd)) {
+			// 	fmt::format_to(std::back_inserter(main_dot), "{}->{}[style=dashed color=gray40];", local_to_global_id(apcmd->get_source()->get_cid()),
+			// 	    local_to_global_id(cmd->get_cid()));
+			// }
```

Comment:
This is obsolete for single-node CDAG prints, isn't it?

Reply:
Well, kind of. At least its a lot more difficult to implement now. But I'm also not sure whether it's all that useful anyway, in particular as it makes layouting the graphs a lot more convoluted.